### PR TITLE
SOL-150080: Adding SAM integration to the docs

### PR DESCRIPTION
**Pull request by solace-awaheed**
## Summary

- Adding SAM Integration Documentation 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Dependency update

## Motivation and Context

Giving the user steps to integrate the MCP with SAM 

## Changes Made

- Updated Documentation 

## Testing

**Test Environment:**
- Go version:
- OS:
- Broker version (if applicable):

**Tests Performed:**
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated (if applicable)
- [ ] E2E tests added/updated (if applicable)
- [ ] Tested locally with `go test -race ./...`
- [ ] Tested with real Solace broker

**Test Coverage:**

-

## Breaking Changes

**Impact:**

**Migration Guide:**

### Code Quality
- [ ] Code follows the [coding standards](../.github/CONTRIBUTING.md#coding-standards)
- [ ] Self-review completed (checked for typos, logic errors, edge cases)
- [ ] Code is well-commented (explains "why", not "what")
- [ ] Complex logic has explanatory comments
- [ ] No commented-out code or debug statements left in

### Security
- [ ] **No credentials in code** (checked all files, commits, and logs)
- [ ] Followed [secure logging rules](../docs/secure-logging-rules.md)
- [ ] Credential-carrying types implement `slog.LogValuer`
- [ ] No security vulnerabilities introduced (checked with golangci-lint/gosec)

### Testing
- [ ] All tests pass locally: `go test -race ./...`
- [ ] No race conditions detected
- [ ] Edge cases covered by tests
- [ ] Error paths tested
- [ ] Test names clearly describe what is being tested

### Documentation
- [ ] README.md updated (if user-facing changes)
- [ ] docs/ updated (if architecture/design changes)
- [ ] godoc comments added/updated for exported functions
- [ ] CHANGELOG.md updated (if applicable)
- [ ] Examples updated (if applicable)

### Git & CI
- [ ] All commits are signed off (DCO: `git commit -s`)
- [ ] Commits have clear, descriptive messages
- [ ] Branch is up to date with main (rebased if needed)
- [ ] No merge conflicts
- [ ] CI checks passing (lint, build, test, E2E)

### Breaking Changes (if applicable)
- [ ] Breaking changes documented in commit message
- [ ] Breaking changes documented in PR description
- [ ] Migration guide provided
- [ ] Deprecated functionality marked with comments and deprecation notices

## Related Issues/PRs

- Related to #
- Depends on #
- Blocks #

---

**For Reviewers:**

**Focus Areas**: 

**Questions**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,6 +217,26 @@ Example query:
...
+```yaml
+# broker-config.yaml
+client_auth:
+  mode: static
+  dev_token: "sam-mcp-dev-token-local-only"
+```
+
+Then in your SAM project, add an MCP-tooled agent that points at this server with the matching bearer token. Full walkthrough: [SAM Integration](docs/sam-integration.md).
+
+Example queries through SAM's WebUI:
```
Comment (SeanWray89) [inline README.md]: SAM's WebUI -> the SAM web UI

File: README.md [hunk 2]
```diff
@@ -215,6 +217,26 @@ Example query:
...
+Once the MCP server is running, configure it to accept a static dev token:
+
+```yaml
+# broker-config.yaml
+client_auth:
+  mode: static
+  dev_token: "sam-mcp-dev-token-local-only"
+```
+
+Then in your SAM project, add an MCP-tooled agent that points at this server with the matching bearer token. Full walkthrough: [SAM Integration](docs/sam-integration.md).
```
Comment (SeanWray89) [inline README.md]: Full walkthrough -> For complete setup instructions, see <link>.

File: docs/sam-integration.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+**5. Verify** — SAM's logs should include:
+
+```
+Initialized MCPToolset for server: url='http://localhost:9090/mcp'
+Agent card tool manifest populated with 14 tools.
+Registered new agent 'SolaceBrokerAgent' in registry.
+```
+
+Open SAM's WebUI (`http://localhost:8000`) and ask the orchestrator something broker-related — it will delegate to `SolaceBrokerAgent`, which calls the MCP tools.
```
Comment (SeanWray89) [inline docs/sam-integration.md]: Open the SAM web UI (`http://localhost:8000`) and ask the orchestrator a broker-related question— it delegates to the `SolaceBrokerAgent`, which calls the MCP tools.

File: docs/sam-integration.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
+# Connecting solace-broker-mcp to Solace Agent Mesh (SAM)
+
+This guide covers only the pieces specific to wiring this MCP server into SAM. For SAM's general MCP support — connection types, tool filtering, TLS/SSL config, environment variable passing — see the upstream [MCP Integration tutorial](https://github.com/SolaceLabs/solace-agent-mesh/blob/main/docs/docs/documentation/developing/tutorials/mcp-integration.md).
```
Comment (SeanWray89) [inline docs/sam-integration.md]: This guide shows how to integrate this MCP server with SAM. For general MCP support in SAM — connection types, tool filtering, TLS/SSL config, environment variable passing — see the upstream [MCP Integration tutorial](https://github.com/SolaceLabs/solace-agent-mesh/blob/main/docs/docs/documentation/developing/tutorials/mcp-integration.md).

File: docs/sam-integration.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+```bash
+# Terminal 1 — this MCP server
+go run ./cmd/server
+
+# Terminal 2 — SAM (auto-discovers the new agent)
+sam run
+```
+
+**5. Verify** — SAM's logs should include:
```
Comment (SeanWray89) [inline docs/sam-integration.md]: the SAM logs should include:

File: docs/sam-integration.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        defaultInputModes: ["text"]
+        defaultOutputModes: ["text", "file"]
+      agent_card_publishing: { interval_seconds: 10 }
+      agent_discovery: { enabled: true }
+      inter_agent_communication:
+        allow_list: ["*"]
+        request_timeout_seconds: 60
+```
+
+**3. Wire the matching token in SAM's `.env`**:
```
Comment (SeanWray89) [inline docs/sam-integration.md]: **3. Add the matching token to the SAM `.env` file**:

File: docs/sam-integration.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
+# Connecting solace-broker-mcp to Solace Agent Mesh (SAM)
+
+This guide covers only the pieces specific to wiring this MCP server into SAM. For SAM's general MCP support — connection types, tool filtering, TLS/SSL config, environment variable passing — see the upstream [MCP Integration tutorial](https://github.com/SolaceLabs/solace-agent-mesh/blob/main/docs/docs/documentation/developing/tutorials/mcp-integration.md).
+
+## Prerequisites
+
+- A working SAM project (`sam init` complete, agents start cleanly)
+- This MCP server runnable locally with valid `broker-config.yaml` + `.env` (see the main [Quickstart](../README.md#quickstart))
```
Comment (SeanWray89) [inline docs/sam-integration.md]: - Ability to run this MCP server locally with configured `broker-config.yaml` and `.env` files. For instructions, see [Quickstart](../README.md#quickstart).

File: docs/sam-integration.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+```
+Initialized MCPToolset for server: url='http://localhost:9090/mcp'
+Agent card tool manifest populated with 14 tools.
+Registered new agent 'SolaceBrokerAgent' in registry.
+```
+
+Open SAM's WebUI (`http://localhost:8000`) and ask the orchestrator something broker-related — it will delegate to `SolaceBrokerAgent`, which calls the MCP tools.
+
+## Auth chain
```
Comment (SeanWray89) [inline docs/sam-integration.md]: Authentication Chain

File: docs/sam-integration.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+```
+SAM agent ─(Bearer SOLACE_BROKER_MCP_TOKEN)→ MCP server ─(basic BROKER_USERNAME/PASSWORD)→ Solace event broker
+```
+
+The bearer token in SAM's `.env` and the `dev_token` in this MCP server's config must be identical.
```
Comment (SeanWray89) [inline docs/sam-integration.md]: The bearer token in the SAM `.env` file and the `dev_token` in the MCP server configuration must be identical.

File: docs/sam-integration.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+## What's covered upstream
```
Comment (SeanWray89) [inline docs/sam-integration.md]: Additional SAM MCP Configuration

File: docs/sam-integration.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+For these, refer to the [SAM MCP Integration tutorial](https://github.com/SolaceLabs/solace-agent-mesh/blob/main/docs/docs/documentation/developing/tutorials/mcp-integration.md):
```
Comment (SeanWray89) [inline docs/sam-integration.md]: The following topics are covered in the [SAM MCP Integration tutorial](https://github.com/SolaceLabs/solace-agent-mesh/blob/main/docs/docs/documentation/developing/tutorials/mcp-integration.md):

- Other connection types (`stdio`, `sse`, Docker)
- Tool filtering (`tool_name`, `allow_list`, `deny_list`)
- TLS/SSL config for self-signed certs (`ssl_config`)
- Passing environment variables to the MCP server process

**Review by bczoma — DISMISSED** [review]
LGTM

File: README.md [hunk 12]
```diff
@@ -232,6 +234,26 @@ Example query:
...
+  mode: static
+  dev_token: "sam-mcp-dev-token-local-only"
+```
+
+Then in your SAM project, add an MCP-tooled agent that points at this server with the matching bearer token. For complete setup instructions, see: [SAM Integration](docs/sam-integration.md).
+
+Example queries through the SAM web UI:
+
+```
+What brokers are configured?
```
Comment (SeanWray89) [inline README.md]: What event brokers are configured?

File: README.md [hunk 13]
```diff
@@ -232,6 +234,26 @@ Example query:
...
+List the queues on broker-one's default VPN.
```
Comment (SeanWray89) [inline README.md]: List the queues on event-broker-one's default VPN.

File: docs/sam-integration.md [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+
+**5. Verify** — the SAM logs should include:
+
+```
+Initialized MCPToolset for server: url='http://localhost:9090/mcp'
+Agent card tool manifest populated with 14 tools.
+Registered new agent 'SolaceBrokerAgent' in registry.
+```
+
+Open the SAM web UI (`http://localhost:8000`) and ask the orchestrator a broker-related question — it delegates to the `SolaceBrokerAgent`, which calls the MCP tools.
```
Comment (SeanWray89) [inline docs/sam-integration.md]: — the orchestrator delegates...

**Review by SeanWray89 — DISMISSED** [review]

**Review by solace-aross — APPROVED** [review]
Approve. Docs-only change and every technical claim checks out against the code (config keys, port 9090, `/mcp` endpoint, the basic-auth broker chain, `go run ./cmd/server`, tool count, Quickstart anchor). The auth-chain diagram is the most useful part — the two-hop bearer→basic distinction is exactly where people get stuck.

Two non-blocking suggestions, fine to fold in here or as follow-up:

- **Example token note.** `sam-mcp-dev-token-local-only` is fine as a labeled local-only placeholder, but the README snippet (unlike the doc's) has no `# change for your environment` comment. Worth a one-line note that it must be rotated and never used outside localhost so nobody copy-pastes it into a shared environment.
- **Verification log block.** The exact strings (`Initialized MCPToolset...`, `Agent card tool manifest populated with 14 tools.`) are SAM's output, not ours, so they can drift silently if SAM changes its format or the tool count shifts. Consider "logs should include lines similar to".

Out of scope: `docs/internal/architecture.md:36` says "11 composite tools" while `tools.yaml` has 10 — pre-existing, worth a separate follow-up.
